site/src/utils/attributes.js

```diff
@@ -0,0 +1,61 @@
+const ATTRS_REGEX = /^(.+)\s+\{(.+)\}$/;
+/**
+ * Extracts attributes from markdown text to be applied to the resulting HTML.
+ * @example
+ * // returns { 
+ * //   text: 'Heading',
+ * //   raw: '<code>Heading</code>',
+ * //   attrs: { id: 'important', class: 'red', test: 'true' },
+ * //   attrstring: 'id="important" class="red" test="true"'
+ * // }
+ * extract_attributes(
+ *   'Heading { #important .red test=true }', 
+ *   '<code>Heading</code> { #important .red test=true }'
+ * );
+ * ```
+ * @param {string} text 
+ * @param {string} raw 
+ */
+export function extract_attributes(text, raw) {
```

Comment:
i wonder do we need such complex attribute logic for the header?

currently we just need to know if `data-toc-ignore` is present right?

Reply:
Fair point, I might have gotten a little carried away 😅 

I can take out the custom class/ID handling. Anything else you think should go?

Reply:
at least for now, all i see is that we just need the `raw` and the `attrs` to fix the heading.

am i missing anything?

Reply:
@tanhauhau I refactored the code to remove the unnecessary `attrstring` and the unused class/id parsing.

We need both `text` and `raw` because the attribute object will appear in both of them. `extract_attributes` will remove the attribute object from `text` and `raw` so that we don't display it in the HTML.